arches/app/models/resource.py

```diff
@@ -343,20 +343,20 @@ def serialize(self):
     def get_node_values(self, node_name):
         """
         Take a node_name (string) as an argument and return a list of values.
-        If an invalid node 
-
-        only string, domain, domain-list, concept, and concept-list datatypes
-        are handled at this time.
+        If an invalid node_name is used, or if multiple nodes with the same
+        name are found, the method returns False.
         """
-        datatype_list = [
-            'string', 'domain', 'domain-list', 'concept', 'concept-list']
+
         nodes = models.Node.objects.filter(
-            name=node_name, datatype__in=datatype_list,graph_id=self.graph_id)
+            name=node_name, graph_id=self.graph_id)
+
         if len(nodes) != 1:
-            print "invalid node name or multiple nodes with the same name"
+            print("invalid node name or multiple nodes with the same name")
             return False
```

Comment:
Instead of returning False here I'd probably throw a couple of different errors to allow end users to disambiguate the error and potentially re-interrogate with a node-id instead.
EG:
```
if len(nodes) == 1:
    run the remaining code here...
elif len(nodes) == 0:
    throw new InvalidNodeNameException(node_name)
elif len(nodes) > 1:
    throw new MultipleNodesFoundException(node_name, nodes)

# example exception class
class MultipleNodesFoundException(Exception):
    def __init__(self, name, nodes):
        self.nodes = nodes
        self.message = "Multiple nodes with the name '%s' were found" % name

class InvalidNodeNameException(Exception):
    def __init__(self, name):
        self.message = "Node with the name '%s' doesn't exist" % name
```